cache/http.go

```diff
@@ -131,49 +112,50 @@ func (h *httpCache) CacheHandler(w http.ResponseWriter, r *http.Request) {
 
 	switch m := r.Method; m {
 	case http.MethodGet:
-		if !h.cache.ContainsFile(cacheItem.absFilePath) {
-			w.WriteHeader(http.StatusNotFound)
+		found, err := h.cache.Get(cacheKey, w)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			h.errorLogger.Printf("get %s: %s", cacheKey, err)
+			return
+		}
+
+		if !found {
 			logResponse(http.StatusNotFound)
 			return
 		}
-		http.ServeFile(w, r, cacheItem.absFilePath)
+
 		logResponse(http.StatusOK)
 	case http.MethodPut:
-		if h.cache.ContainsFile(cacheItem.absFilePath) {
-			h.discardUpload(w, r.Body)
-			logResponse(http.StatusOK)
+		if r.ContentLength == -1 {
```

Comment:
We might want to check for `Transfer Encoding: chunked` and simply disallow it. This would seem to be the most likely cause of a missing `Content Length` header.

Reply:
Mmm, how would that be better than complaining about the `Content-Length` header directly?